### PR TITLE
feat: add Java 17 support

### PR DESCRIPTION
**Pull request by marcellorinaldo**
**Maven plugins updates:**

- `bnd-baseline-maven-plugin`: 3.4.0 -> 6.3.1
- `maven-bundle-plugin`: 3.3.0 -> 5.1.8
- `maven-compiler-plugin`: 3.3 -> 3.10.1
- `target-platform-configuration`: 1.6.0 -> 3.0.0
- `tycho-compiler-plugin`: 1.6.0 -> 3.0.0
- `tycho-maven-plugin`: 1.6.0 -> 3.0.0
- `tycho-p2-extras-plugin`: 0.26.0 -> 3.0.0
- `tycho-source-plugin`: 1.6.0 -> 3.0.0
- `tycho-sourceref-jgit`: 1.6.0 -> 3.0.0

**Dependencies updates:**

Tests
- `junit`: 4.12 -> 4.13.2
- `mockito-core`: 1.10.19 -> 4.8.1
- `objenesis`: 2.4 -> 3.3

**New dependencies:**

- `byte-buddy`: 1.12.18
- `byte-buddy-agent`: 1.12.18

**Summary of changes**

- Target platform: renamed `org.apache.log4j2-api-config` to `log4j2-api-config` since `.` and `-` are not allowed simultaneously in jar names
- Target platform: changed some steps in the definitions of `maven-bundle-plugin` configs
- Target platform & Kura: refactor of poms to have all the versions specified as properties
- Kura: updated manifests for correct log4j versioning
- Kura: updated manifests to remove deprecated required exec header and replaced with capability osgi.ee filter to allow both Java 1.8 and Java 17 execution environments
- Test: migration to mockito 4
- Build: added profile for separating Java 8 build steps from Java 17 ones based on the running jvm
- Build: added `buildAll` property in place of `activeBydefault` for the distrib since the latter is activated only when no other profiles are selected. That is no more the case since a profile between `java8` and `java17` is always active

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
@MMaiero Since I've updated `mockito-core` to 4.8.1, do we need a CQ for the required additional test dependencies?

```xml
<artifactItem>
    <groupId>net.bytebuddy</groupId>
    <artifactId>byte-buddy</artifactId>
    <version>1.12.18</version>
</artifactItem>
<artifactItem>
    <groupId>net.bytebuddy</groupId>
    <artifactId>byte-buddy-agent</artifactId>
    <version>1.12.18</version>
</artifactItem>
```

They are distributed with the Apache 2.0 License [ref](https://github.com/raphw/byte-buddy/blob/master/LICENSE).

**Comment by marcellorinaldo** [comment]
> @MMaiero Since I've updated `mockito-core` to 4.8.1, do we need a CQ for the required additional test dependencies?
> 
> ```
> <artifactItem>
>     <groupId>net.bytebuddy</groupId>
>     <artifactId>byte-buddy</artifactId>
>     <version>1.12.18</version>
> </artifactItem>
> <artifactItem>
>     <groupId>net.bytebuddy</groupId>
>     <artifactId>byte-buddy-agent</artifactId>
>     <version>1.12.18</version>
> </artifactItem>
> ```
> 
> They are distributed with the Apache 2.0 License [ref](https://github.com/raphw/byte-buddy/blob/master/LICENSE).

https://www.eclipse.org/projects/handbook/#ip-third-party-test

**Comment by marcellorinaldo** [comment]
> > @MMaiero Since I've updated `mockito-core` to 4.8.1, do we need a CQ for the required additional test dependencies?
> > ```
> > <artifactItem>
> >     <groupId>net.bytebuddy</groupId>
> >     <artifactId>byte-buddy</artifactId>
> >     <version>1.12.18</version>
> > </artifactItem>
> > <artifactItem>
> >     <groupId>net.bytebuddy</groupId>
> >     <artifactId>byte-buddy-agent</artifactId>
> >     <version>1.12.18</version>
> > </artifactItem>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > They are distributed with the Apache 2.0 License [ref](https://github.com/raphw/byte-buddy/blob/master/LICENSE).
> 
> https://www.eclipse.org/projects/handbook/#ip-third-party-test

The dependencies seems ok using the [dash-licenses tool](https://github.com/eclipse/dash-licenses). Here the output:

```
echo "maven/mavencentral/net.bytebuddy/byte-buddy/1.12.18" | java -jar org.eclipse.dash.licenses-0.0.1-20221107.065029-602.jar -
[main] INFO Querying Eclipse Foundation for license data for 1 items.
[main] INFO Found 1 items.
[main] INFO Vetted license information was found for all content. No further investigation is required.

echo "maven/mavencentral/net.bytebuddy/byte-buddy-agent/1.12.18" | java -jar org.eclipse.dash.licenses-0.0.1-20221107.065029-602.jar -
[main] INFO Querying Eclipse Foundation for license data for 1 items.
[main] INFO Found 1 items.
[main] INFO Vetted license information was found for all content. No further investigation is required.
```

**Comment by MMaiero** [comment]
@GregoryIvo Can you please have a review of this PR? Thanks

**Comment by GregoryIvo** [comment]
Please check all the Copywrite headers please

**Comment by GregoryIvo** [comment]
Update notice file

**Comment by GregoryIvo** [comment]
Check if licences are okay with the new version of GWT bootstrap

**Comment by marcellorinaldo** [comment]
> Check if licences are okay with the new version of GWT bootstrap

Licenses are ok. Output of the Eclipse license tool:

```
org.gwtbootstrap3/gwtbootstrap3/1.0.1
[main] INFO Querying Eclipse Foundation for license data for 1 items.
[main] INFO Found 1 items.
[main] INFO Vetted license information was found for all content. No further investigation is required.
```

**Comment by marcellorinaldo** [comment]
Waiting for CQ response for the following dependencies:

- [x] `gwt-user:2.10.0` https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4958
- [x] `log4j-core:2.19.0` https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/5009
- [x] `log4j-slf4j-impl:2.19.0` https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/5010

File: kura/examples/test/org.eclipse.kura.example.ble.tisensortag.dbus.test/src/test/java/org/eclipse/kura/example/ble/tisensortag/dbus/TiSensorTagTest.java [hunk 1]
```diff
@@ -377,7 +377,7 @@ public void testDisableAccelerometerCc2650Fail()
 
         tag.disableAccelerometer();
 
-        verify(bch, times(0)).writeValue(anyObject());
+        verify(bch, times(0)).writeValue(any());
```
Comment (GregoryIvo) [inline kura/examples/test/org.eclipse.kura.example.ble.tisensortag.dbus.test/src/test/java/org/eclipse/kura/example/ble/tisensortag/dbus/TiSensorTagTest.java:380]: is any() only compliant with java17?  I will check soon.

**Review by GregoryIvo — APPROVED** [review]
I checked for consistency between manifests. The new version changes appear to be consistent in all POMS.
It also builds successfully on my end with JDK.

LGTM!

**Comment by MMaiero** [comment]
Is the PR ready to merge?
@nicolatimeus ok for you on the changes?

**Comment by marcellorinaldo** [comment]
> Is the PR ready to merge?

Just waiting for the full QA results

**Comment by nicolatimeus** [comment]
There may be a problem with conventional commits if we merge this as is: do we have to make a PR with `build` type for each of the updated/added dependencies so that they end up in the release notes separately?

File: kura/emulator/org.eclipse.kura.emulator.clock/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -14,5 +14,5 @@ Import-Package: org.eclipse.kura;version="[1.0,2.0)",
  org.osgi.framework;version="1.5.0",
  org.osgi.service.component;version="1.2.0",
  org.osgi.service.event;version="1.3.0",
- org.slf4j;version="1.6.4"
```
Comment (nicolatimeus) [inline kura/emulator/org.eclipse.kura.emulator.clock/META-INF/MANIFEST.MF]: What is the reason for this change? previously this import allows all `org.slf4j` versions from 1.6.4 onwards, this reduces the range from 1.7.36 to 1.8.0 excluded.

Comment (marcellorinaldo) [reply kura/emulator/org.eclipse.kura.emulator.clock/META-INF/MANIFEST.MF]: I restricted the versions to max. 1.8 because we are using `log4j-slf4j-impl`, which only supports slf4j < 1.8 (https://github.com/apache/logging-log4j2/blob/release-2.x/RELEASE-NOTES.md#apache-log4j-2190-release-notes). No particular reason for the lower bound enforcement.

**Comment by marcellorinaldo** [comment]
Deps update are made here: https://github.com/eclipse/kura/pull/4259

**Comment by marcellorinaldo** [comment]
Should be ready to merge.

**Comment by MMaiero** [comment]
@nicolatimeus ok for you to merge?

File: kura/org.eclipse.kura.wire.h2db.component.provider/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -6,7 +6,7 @@ Bundle-Version: 2.3.0.qualifier
 Bundle-Vendor: Eclipse Kura
 Bundle-License: Eclipse Public License v2.0
 Bundle-Category: Kura Wires
-Import-Package: org.apache.logging.log4j;version="2.8.2",
+Import-Package: org.apache.logging.log4j;version="2.19.0",
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.wire.h2db.component.provider/META-INF/MANIFEST.MF]: The considerations for the `org.slf4j` import should apply also for this package

File: kura/org.eclipse.kura.wire.script.filter.provider/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -12,6 +12,7 @@ Import-Package: javax.script,
  org.osgi.service.component;version="1.2.0",
  org.osgi.service.wireadmin;version="1.0.1",
  org.slf4j;version="1.7.0"
+Require-Bundle: org.eclipse.osgi
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.wire.script.filter.provider/META-INF/MANIFEST.MF]: Why is this needed?

Comment (marcellorinaldo) [reply kura/org.eclipse.kura.wire.script.filter.provider/META-INF/MANIFEST.MF]: Not needed, maybe accidentally included using the IDE.

File: .gitignore [hunk 5]
```diff
@@ -37,3 +37,4 @@ bin/
 *.iml
 maven.deps
 DEPENDENCIES
+lib/
```
Comment (nicolatimeus) [inline .gitignore]: Does this ignore all folders named `lib` of the entire reposistory? If you want to ignore a specific lib folder better add a new .gitignore file in its parent directory

**Review by nicolatimeus — APPROVED** [review]
